lib/impure/nre.nim

```diff
@@ -578,7 +578,7 @@ iterator findIter*(str: string, pattern: Regex, start = 0, endpos = int.high): R
       # either the end of the input or the string
       # cannot be split here - we also need to bail
       # if we've never matched and we've already tried to...
-      if offset >= strlen or neverMatched:
+      if flags == 0 or offset >= strlen or neverMatched: # All matches found
```

Comment:
I loose the restriction by adding `flags == 0`(`flags` is equals to `options` in pcre), I'm not sure about other parts.